### PR TITLE
Migrate to null-safety

### PR DESCRIPTION
**Pull request by hoc081098**

### CODE REVIEW

**Comment by gunasekharravilla** [comment]
looks very busy with the errors from yesterday 😂😂

File: MobileApp/datn/lib/data/mappers.dart [hunk 1]
```diff
@@ -289,7 +295,7 @@ Movie movieDetailResponseToMovie(MovieDetailResponse res) {
       final categoriesBuilder = b.categories
         ..safeReplace(res.categories.map(categoryResponseToCategory));
 
-      return b
+      b
```
Comment (gunasekharravilla) [inline MobileApp/datn/lib/data/mappers.dart:298]: I'm getting error in line 352 like this

```
{required String reservationId}
The argument type 'String?' can't be assigned to the parameter type 'String'.dart(argument_type_not_assignable)
```

Comment (hoc081098) [reply MobileApp/datn/lib/data/mappers.dart:298]: it has already been fixed

**Comment by gunasekharravilla** [comment]
I observed one issue now after migrating to null-safety
 
that is after successfully payment completion reservation timer is running after the timer expires it's giving alert and meanwhile app is trying to reconnect the server number of times in a loop

```` I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
I/flutter ( 9162): [ReservationRepositoryImpl] connected
I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null````

**Comment by gunasekharravilla** [comment]
> I observed one issue now after migrating to null-safety
> 
> that is after successfully payment completion reservation timer is running after the timer expires it's giving alert and meanwhile app is trying to reconnect the server number of times in a loop
> 
> ```
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
> I/flutter ( 9162): [ReservationRepositoryImpl] connected
> I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
> I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
> I/flutter ( 9162): [ReservationRepositoryImpl] connected
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
> I/flutter ( 9162): [ReservationRepositoryImpl] connected
> I/flutter ( 9162): [ReservationRepositoryImpl] disconnected ping timeout null
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnect_attempt 1
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnecting 1
> I/flutter ( 9162): [ReservationRepositoryImpl] reconnect 1
```


I forgot to import `import 'package:rxdart_ext/rxdart_ext.dart';`  this may be reason ?

**Comment by hoc081098** [comment]
@gunasekharravilla thanks, that is my mistake 😂

_Sent from my Redmi  7A using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by gunasekharravilla** [comment]
oh no problem fix it when possible

**Comment by gunasekharravilla** [comment]
`dart migrate --apply-changes` or
`dart migrate --apply-changes --skip-import-check`

use this command it may help you to migrate to null safety easily

**Comment by hoc081098** [comment]
> `dart migrate --apply-changes` or
> `dart migrate --apply-changes --skip-import-check`
> 
> use this command it may help you to migrate to null safety easily

This command usually inserts '?', that is not suitable for some cases :)

**Comment by gunasekharravilla** [comment]
When I tried it applies only 1 change in main.dart 
For signOut() function appended late in front of the function

**Pull request merged**